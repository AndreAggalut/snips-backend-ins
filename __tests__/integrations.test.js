require('dotenv').config();
const request = require('supertest');
const app = require('../app');
const dbInit = require('../db/init');
const db = require('../db');

beforeAll(async () => {
  await dbInit.createTables();
  await dbInit.seedAuthors();
  await dbInit.seedSnippets();
});

describe('Snippets', () => {
  describe('GET /api/snippets', () => {
    it('should get all of the snips', async () => {
      // test the /api/snips route
      const response = await request(app).get('/api/snippets');
      // expect two rows of data
      expect(response.body.length).toBe(2);
      // no errors
      expect(response.error).toBeFalsy();
      // status code should be 200
      expect(response.status).toBe(200);
      // matches the data directly
      expect(response.body).toMatchSnapshot();
    });
  });
});

// cleanup
afterAll(() => {
  // close the db pool
  db.end();
});
