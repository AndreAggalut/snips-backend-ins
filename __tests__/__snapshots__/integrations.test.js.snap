// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippets GET /api/snippets should get all of the snips 1`] = `
Array [
  Object {
    "author": "Dandy",
    "code": "const america = 1776",
    "description": "I declared a const",
    "favorites": 0,
    "id": 1,
    "language": "JavaScript",
    "title": "freedom",
  },
  Object {
    "author": "Scott",
    "code": "4 + 4",
    "description": "This is how you add",
    "favorites": 0,
    "id": 2,
    "language": "Algebra",
    "title": "addition",
  },
]
`;
